src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -256,6 +249,61 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         _AttachedHandlers(*this, nullptr);
     }
 
+    TerminalConnection::ITerminalConnection ControlCore::Connection()
+    {
+        return _connection;
+    }
+
+    // Method Description:
+    // - Setup our event handlers for this connection. If we've currently got a
+    //   connection, then this'll revoke the existing connection's handlers.
+    void ControlCore::Connection(const TerminalConnection::ITerminalConnection& newConnection)
+    {
+        if (_connection)
+        {
+            _connectionOutputEventRevoker.revoke();
+        }
+
+        // Subscribe to the connection's disconnected event and call our connection closed handlers.
+        _connectionStateChangedRevoker = newConnection.StateChanged(winrt::auto_revoke, [this](auto&& /*s*/, auto&& /*v*/) {
+            _ConnectionStateChangedHandlers(*this, nullptr);
+        });
+
+        // Fire off a connection state changed notification, to let our hosting
+        // app know that we're in a different state now.
+        if (newConnection.State() != _connection.State())
+        {
+            _ConnectionStateChangedHandlers(*this, nullptr);
+        }
+
+        // Get our current size in rows/cols, and hook them up to
+        // this connection too.
+        {
+            const auto vp = _terminal->GetViewport();
+            const auto width = vp.Width();
+            const auto height = vp.Height();
+
+            newConnection.Resize(height, width);

```

Comment:
Ah, but I just realized that `newConnection` is probably in the unconnected state at this point, right? So `Resize` wouldn't do much anyways... Although I still think that my comment regarding the duplication is a minor concern at least. BTW does this function even work if you pass it a already connected `newConnection`? If not, we should probably assert on that.

Reply:
Yeah, I get that, but isn't it kind of confusing to have that logic in more than one place? It seems to me like, if there's one place that can instantly override the connection's size, then the connection might not be the right thing to carry the size to begin with.

Reply:
I mean, yea, it's kinda confusing for sure. Cause basically

1. the app starts by saying "I'm gonna make a control that's CxRs big"
2. We build a TerminalSettings that says as much
3. We use those TerminalSettings to build a connection, and we initially tell that connection "hey you're gonna be CxR big".
4. We build the actual control. We were smart enough to give it enough space to be CxR big.
   * For something like a pane, the control might end up resizing the connection (before starting it), because the pane used less space
5. We start the connection. using what it knows, it starts at the size that it's got stored inside itself (so it doesn't need to go _ask_ how big it should start at. 

So like, I suppose theoretically we could leave out the part where TerminalPage passes an initial rows / columns to a connection. We'd need to always rely on the control to resize the connection to the appropriate size. I suppose that's a refactoring we could do. But probably not for this PR?